placement/algo/sharded_helper.go

```diff
@@ -818,6 +818,28 @@ func markShardAvailable(p placement.Placement, instanceID string, shardID uint32
 	return p, nil
 }
 
+func tryCleanupShardState(
+	p placement.Placement,
+	opts placement.Options,
+) (placement.Placement, error) {
+	if opts.ShardStateType() == placement.SimpleShardState {
```

Comment:
Why is it a problem to mark shards as available for transitioning shards? Thought that was what the old logic does..

Reply:
Can you elaborate on "hard to make MarkShardAsAvailable" calls?

Reply:
Looked at this again, so basically you are using the current placement algorithm to generate placements, but for simple shard states, you'll immediately mark them as available (i.e., removing leaving shards and marking initializing shards as available) before returning them to the caller. Is that correct?

If so, perhaps change the option to a boolean `AllowTransitionalStates` option instead of defining `ShardStateType`, and will mark all states as available if the option is false? That IMO feels more intuitive. Right now the `SimpleShardState` is still using `Available` state, so technically it's not a different shard state and that's a bit confusing.

Reply:
Sure, would it be better to call it `UseSingleShardState`?

Reply:
I'd rather say NoShardState or IgnoreShardState, cause it does not matter to user if there is one shard state or more, user should not check the shard state at all, we may also use Available or other states in implementation.

Reply:
Yeah however we are returning a placement to the user who can technically access the shards and see the shard states as available, so unless we can literally set the shard state to `None`, I think calling it `NoShardState` would be inaccurate. I'm fine with `IgnoreShardStates` though.